lib/devices_api_web/auth/jwt_auth_plug.ex

```diff
@@ -0,0 +1,42 @@
+defmodule DevicesApiWeb.Auth.JwtAuthPlug do
+  @moduledoc """
+  Enables token verification
+  """
+
+  import Plug.Conn
+
+  alias DevicesApiWeb.Auth.JwtToken
+
+  @doc "Initializes plug"
+  @spec init(opts :: Keyword.t()) :: list()
+  def init(opts) do
+    opts
+  end
+
+  @doc "Verifies a token"
+  @spec call(conn :: Plug.Conn.t(), opts :: list()) :: {:unauthorized, String.t()} | Plug.Conn.t()
+  def call(conn, _opts) do
+    with {:ok, token} <- extract_token(conn),
+         {:ok, claims} <- JwtToken.verify_signature(token),
+         {:ok, claims} <- JwtToken.verify_claims(claims) do
+      conn
+      |> assign(:current_user, claims)
+    else
+      {:error, _ = message} -> unauthorize(conn, message)
+    end
+  end
+
+  defp extract_token(conn) do
+    case get_req_header(conn, "authorization") do
+      ["Bearer " <> token] -> {:ok, token}
+      _ -> {:error, "No authorization token found!"}
+    end
+  end
```

Comment:
Aqui tem um problema MUITO SUTIL...

headers são "multi-valued" pela especificação de HTTP. Da forma como está aqui, eu consigo quebrar sua interface se fizer assim:

```
authorization: Bearer jwt; Bearer jwt
```

O ideal em tratar headers que queremos somente o primeiro valor é tratar com um match de lista ali: 

```suggestion
  defp extract_token(conn) do
    case get_req_header(conn, "authorization") do
      ["Bearer " <> token | _] -> {:ok, token}
      _ -> {:error, "No authorization token found!"}
    end
  end
```

Cuidado com retornos com mensagens também. Isso deveria ser exclusivo da view e não do controller.